packages/@react-aria/interactions/src/utils.ts

```diff
@@ -118,8 +118,10 @@ export function useSyntheticBlurEvent<Target = Element>(onBlur: (e: ReactFocusEv
       stateRef.current.observer = new MutationObserver(() => {
         if (stateRef.current.isFocused && target.disabled) {
           stateRef.current.observer.disconnect();
-          target.dispatchEvent(new FocusEvent('blur'));
-          target.dispatchEvent(new FocusEvent('focusout', {bubbles: true}));
+          // added `relatedTarget: document.activeElement` to keep focus on activeElment for Firefox DatePicker closing popover with minValue
+          let relatedTarget = target === document.activeElement ? null : {relatedTarget: document.activeElement};
```

Comment:
A test failed when blurring the activeElement. Blur only needs to retain focus on the activeElement when it isn't the element being blurred.

Reply:
which test failed?

Reply:
It was "should fire onBlur when a focused element is disabled" within packages/@react-aria/interactions/test/useFocusWithin.test.js

Reply:
hmm wonder if JSDOM is just not updating the active element correctly when a button becomes disabled? Otherwise I'm not sure how the button would still be focused. Setting it to null seems right.

Only thing I'd change here is that the `relatedTarget` should always be set, even when it's null. So:

```suggestion
          let relatedTarget = target === document.activeElement ? null : document.activeElement;
```

And then, always add the `relatedTarget` property to the object below.